src/Raven.Server/Documents/Handlers/Processors/Indexes/AbstractIndexHandlerProcessorForTerms.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client.Http;
+using Raven.Server.Documents.Queries;
+using Raven.Server.Json;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Web;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers.Processors.Indexes
+{
+    internal abstract class AbstractIndexHandlerProcessorForTerms<TRequestHandler, TOperationContext> : AbstractHandlerProcessor<TRequestHandler, TOperationContext>
+        where TRequestHandler : RequestHandler
+        where TOperationContext : JsonOperationContext
+    {
+        protected AbstractIndexHandlerProcessorForTerms([NotNull] TRequestHandler requestHandler, [NotNull] JsonContextPoolBase<TOperationContext> contextPool)
+            : base(requestHandler, contextPool)
+        {
+        }
+
+        protected abstract ValueTask<TermsQueryResultServerSide> GetTerms(TransactionOperationContext context, string indexName, string field, string fromValue, int pageSize);
+
+        public override async ValueTask ExecuteAsync()
+        {
+            using (RequestHandler.ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            {
+                var field = RequestHandler.GetQueryStringValueAndAssertIfSingleAndNotEmpty("field");
+                var indexName = RequestHandler.GetQueryStringValueAndAssertIfSingleAndNotEmpty("name");
+                var fromValue = RequestHandler.GetStringQueryString("fromValue", required: false) ?? "";
+                var pageSize = RequestHandler.GetIntValueQueryString("pageSize", required: false) ?? int.MaxValue;
+
+                var terms = await GetTerms(context, indexName, field, fromValue, pageSize);
+
+                if (terms.NotModified)
+                    return;
+
+
+                await using (var writer = new AsyncBlittableJsonTextWriter(context, RequestHandler.ResponseBodyStream()))
+                {
+                    writer.WriteTermsQueryResult(context, terms);
+                }
+            }
+        }
+
+        public class GetTermsCommand : RavenCommand<TermsQueryResultServerSide>
```

Comment:
we have this command in client API?

Reply:
Yes, but the return type is different

Reply:
inherit from the command in the client API and override the `SetResponse`

Reply:
done